js/src/credentialRequest/types/LdpVcCredentialRequest.js

```diff
@@ -0,0 +1,58 @@
+const {ProofType} = require("../../constants/ProofType");
+const CredentialRequest = require("../credentialRequest");
+
+class LdpVcCredentialRequest extends CredentialRequest {
+    constructor(accessToken, issuerMetaData, proof) {
+        super();
+        this.accessToken = accessToken;
+        this.issuerMetaData = issuerMetaData;
+        this.proof = proof;
+    }
+
+    constructRequest() {
+        const request = {
+          requestBody: this.generateRequestBody(),
+          requestHeader: {
+            'Authorization': `Bearer ${this.accessToken}`,
+            'Content-Type': 'application/json',
+          },
+        };
+        return request;
+      }
+
+    generateRequestBody() {
+        try {
+            const requestBody = new CredentialRequestBody(
+              this.issuerMetaData.credentialFormat,
+              new CredentialDefinition(this.issuerMetaData.credentialType),
+              {
+                proof_type: ProofType.JWT,
+                jwt: this.proof,
+              },
+            ).toJson();
+            return requestBody;
+        } catch (error) {
+            throw new Error('DownloadFailedError: requestGenerationFailed');
+        }
+    }
+}
+
+class CredentialRequestBody {
+    constructor(format, credentialDefinition, proof) {
+        this.format = format;
+        this.credential_definition = credentialDefinition;
+        this.proof = proof;
+    }
+    toJson() {
+        return JSON.stringify(this);
```

Comment:
we are performing stringification here but the method name says like we are converting to JSON format, can we modify method name accordingly?